java-checks/src/main/java/org/sonar/java/checks/ImmediateReverseBoxingCheck.java

```diff
@@ -112,10 +117,19 @@ private void visitMethodInvocationTree(MethodInvocationTree mit) {
     }
   }
 
-  private void checkForUselessUnboxing(Type targetType, Tree reportTree, ExpressionTree arg0) {
+  private void checkForUselessUnboxing(Type targetType, Tree reportTree, ExpressionTree arg0, Tree originalTree) {
     Type argType = arg0.symbolType();
     if (argType.is(targetType.fullyQualifiedName())) {
-      reportIssue(reportTree, String.format("Remove the boxing to \"%s\"; The argument is already of the same type.", argType.name()));
+      ((InternalJavaIssueBuilder) ((DefaultJavaFileScannerContext) context).newIssue())
+        .forRule(this)
+        .onTree(reportTree)
+        .withMessage(String.format("Remove the boxing to \"%s\"; The argument is already of the same type.", argType.name()))
+        .withQuickFix(() -> JavaQuickFix.newQuickFix("Remove the boxing")
```

Comment:
it seems to me that the following code  is duplicated:
```java
() -> JavaQuickFix.newQuickFix("Remove the boxing")
        .addTextEdit(
          JavaTextEdit.replaceTextSpan(AnalyzerMessage.textSpanBetween(originalTree, true, boxingArg, false), ""),
          JavaTextEdit.replaceTextSpan(AnalyzerMessage.textSpanBetween(boxingArg, false, originalTree, true), ""))
        .build()
```
it can maybe be extracted into a dedicated static method building a quickfix with 2 args

Reply:
Indeed, nice catch!